### DISCUSSION TITLE
How to set groq API as default?

### DISCUSSION THREAD

**Discussion by aborruso** [gptme/gptme#230] [Q&A] [answered]
Hi,
I have set in my config `GROQ_API_KEY = "gsk_CIhrpxxxxxxoXQLVufJ07jBXPEIRHuz"`.

I want it as default. If I run simply `gptme` I have

```
No API key set for OpenAI, Anthropic, or OpenRouter.
You can get one at:
 - OpenAI: https://platform.openai.com/account/api-keys
 - Anthropic: https://console.anthropic.com/settings/keys
 - OpenRouter: https://openrouter.ai/settings/keys

Your OpenAI, Anthropic, or OpenRouter API key:
```

How to set groq as default?

If I set 

```
export GROQ_API_KEY=gsk_CIhrpxxxxxxoXQLVufJ07jBXPEIRHuz
```

and then open it `gptme -m groq/llama-3.1-70b-versatile hello`, after my first sentence `A cat name`, I have

```
openai.BadRequestError: Error code: 400 - {'error': {'message': "'messages.2' : for 'role:assistant' the following must be satisfied[('messages.2.content' : value must be a string)]", 'type': 'invalid_request_error'}}
```

Thank you

**Comment by ErikBjare** [comment, marked as answer]
Also reported here: https://github.com/ErikBjare/gptme/issues/227

Will probably fix it tomorrow.

You could set it as default by editing the config file: https://gptme.org/docs/config.html

**Reply by aborruso** [reply]
Hi,
I have set in this way

![image](https://github.com/user-attachments/assets/31b64f32-eb4a-4751-aea3-73b6c8288c0a)


It seems to work. When I start it, I have a warning about Unknown model. Is it a real problem?

Thank you for the new release

![image](https://github.com/user-attachments/assets/325eaeb6-db58-425c-b102-4713d0217021)

**Reply by ErikBjare** [reply]
You can safely ignore that warning, it just means we don't have metadata about the model built into gptme (like token cost, max context length, etc.)

I should probably remove it.

**Comment by TimeToBuildBob** [comment]
**Update (2026-07-03):** Both issues here were fixed in the gptme codebase a while ago — wanted to add a concrete reference for anyone landing on this page.

**Setting groq as the default model**

Add this to your `gptme.toml` (in your workspace root, or `~/.config/gptme/config.toml` for global config):

```toml
model = "groq/llama-3.3-70b-versatile"
```

The `groq/` prefix tells gptme to route through LiteLLM's Groq provider. Your `GROQ_API_KEY` environment variable is picked up automatically.

**The 400 `content must be a string` error**

This was fixed in commit [`84c83166`](https://github.com/gptme/gptme/commit/84c83166368505d2969fc79dd9951074d2d7228f) (Dec 2024) — gptme now transforms message content to a string before sending to Groq, since Groq doesn't accept the array format that other providers support. The fix is in all releases from v0.18 onward.

If you're on an older install: `pip install --upgrade gptme` should get you past it.

For a current list of supported Groq models, see the [providers docs](https://gptme.org/docs/providers.html#groq) — the Groq section was added in late 2024.
